### PR TITLE
Update app to application set

### PR DESCRIPTION
**Pull request by sitilge**
Update app to application set.

### CODE REVIEW

**Comment by jeandemeusy** [comment]
How does each netwatcher instance receive a different `API_HOST` and `API_KEY` as environment variable ?

**Review by jeandemeusy — APPROVED** [review]

**Comment by ausias-armesto** [comment]
> @jeandemeusy yes. There has to be a secret for each netwatcher called `netwatcher-*`.

Why the secrets are not created in this PR, the feature should be functional

**Comment by ausias-armesto** [comment]
> > @jeandemeusy yes. There has to be a secret for each netwatcher called `netwatcher-*`.
> 
> Why the secrets are not created in this PR, the feature should be functional

Now I understand that the nodes are still not created. I will create the node for staging. This Application set should put in staging only, as in production we don't have the nodes either.

File: staging/netwatcher/application-set.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                AGG_POST: http://production-aggregator.production:8080/aggregator/peers
+                AGG_BALANCE: http://production-aggregator.production:8080/aggregator/balances
+                LAT_COUNT: 10
+                MOCK_MODE: True
+              envFrom:
+                - secretRef:
+                    name: {{ name }}
+    destination:
+      server: https://kubernetes.default.svc
+      namespace: production
```
Comment (ausias-armesto) [inline staging/netwatcher/application-set.yaml:46]: Is this namespece correct?

File: staging/netwatcher/application-set.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - list:
+        elements:
+          - name: netwatcher-1
+          - name: netwatcher-2
+          - name: netwatcher-3
+          - name: netwatcher-4
+  template:
+    metadata:
+      name: {{ name }}
+      namespace: production
```
Comment (ausias-armesto) [inline staging/netwatcher/application-set.yaml:17]: Same here? Typo?

File: staging/netwatcher/application-set.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: netwatcher
+  namespace: production
```
Comment (ausias-armesto) [inline staging/netwatcher/application-set.yaml:5]: and here

**Review by ausias-armesto — CHANGES_REQUESTED** [review]

**Review by ausias-armesto — APPROVED** [review]

**Pull request merged**